### PR TITLE
Home 3

### PR DESCRIPTION
**Pull request by IgorPrihodko**

### CODE REVIEW

File: src/main/java/controller/InitServlet.java [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+
+@WebServlet(value = "/", loadOnStartup = 1)
+public class InitServlet extends HttpServlet {
+
+    @Override
+    public void init() throws ServletException {
+        AllUsersServlet allUsersServlet = new AllUsersServlet();
+        allUsersServlet.initialUser();
```
Comment (SevostyanovGerman) [inline src/main/java/controller/InitServlet.java]: just init test data here, without use other servlets

File: src/main/java/controller/product/AllProductServlet.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+import java.io.IOException;
+import java.util.List;
+
+@WebServlet(value = "/products")
+public class AllProductServlet extends HttpServlet {
+
+    private static final ProductDao productDao = ProductServiceFactory.getInstance();
+    private List<Product> allProducts = productDao.getAll();
+
+    public void initialProduct() {
```
Comment (SevostyanovGerman) [inline src/main/java/controller/product/AllProductServlet.java]: move to InitServlet

File: src/main/java/controller/product/DeleteProductServlet.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+@WebServlet(value = "/deleteProduct")
+public class DeleteProductServlet extends HttpServlet {
+
+    private ProductDao productDao = ProductDaoFactory.getInstance();
```
Comment (SevostyanovGerman) [inline src/main/java/controller/product/DeleteProductServlet.java]: static final?

File: src/main/java/controller/product/EditProductServlet.java [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+@WebServlet(value = "/editProduct")
+public class EditProductServlet extends HttpServlet {
+
+    private ProductDao productDao = ProductDaoFactory.getInstance();
```
Comment (SevostyanovGerman) [inline src/main/java/controller/product/EditProductServlet.java]: -=-

File: src/main/java/controller/product/EditProductServlet.java [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+            }
+            if (price <= 0) {
+                req.setAttribute("error", "Price can not be 0 or less! Try another.");
+            } else {
+                req.setAttribute("price", price);
+            }
+            req.getRequestDispatcher("editProduct.jsp").forward(req, resp);
+            resp.sendRedirect("/editProduct");
+        }
+        productDao.getById(id).setTitle(title);
```
Comment (SevostyanovGerman) [inline src/main/java/controller/product/EditProductServlet.java]: O-O why 3 connect to DB, and why u do that via dao?

File: src/main/java/controller/user/DeleteUserServlet.java [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+@WebServlet(value = "/deleteUser")
+public class DeleteUserServlet extends HttpServlet {
+
+    private UserDao userDao = UserServiceFactory.getInstance();
```
Comment (SevostyanovGerman) [inline src/main/java/controller/user/DeleteUserServlet.java]: -=-

File: src/main/java/controller/user/EditUserServlet.java [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+@WebServlet(value = "/editUser")
+public class EditUserServlet extends HttpServlet {
+
+    private UserDao userDao = UserServiceFactory.getInstance();
```
Comment (SevostyanovGerman) [inline src/main/java/controller/user/EditUserServlet.java]: Service may be? Controllers don't work with dao

**Review by SevostyanovGerman — CHANGES_REQUESTED** [review]

File: src/main/java/controller/user/EditUserServlet.java (hunk 7 above)
Comment (IgorPrihodko) [reply src/main/java/controller/user/EditUserServlet.java]: fixed

File: src/main/java/controller/user/DeleteUserServlet.java (hunk 6 above)
Comment (IgorPrihodko) [reply src/main/java/controller/user/DeleteUserServlet.java]: fixed

File: src/main/java/controller/product/DeleteProductServlet.java (hunk 3 above)
Comment (IgorPrihodko) [reply src/main/java/controller/product/DeleteProductServlet.java]: fixed

File: src/main/java/controller/product/EditProductServlet.java (hunk 4 above)
Comment (IgorPrihodko) [reply src/main/java/controller/product/EditProductServlet.java]: fixed

File: src/main/java/controller/product/EditProductServlet.java (hunk 5 above)
Comment (IgorPrihodko) [reply src/main/java/controller/product/EditProductServlet.java]: fixed

File: src/main/java/controller/product/AllProductServlet.java (hunk 2 above)
Comment (IgorPrihodko) [reply src/main/java/controller/product/AllProductServlet.java]: done

File: src/main/java/controller/InitServlet.java (hunk 1 above)
Comment (IgorPrihodko) [reply src/main/java/controller/InitServlet.java]: rebuilt

File: src/main/java/controller/product/EditProductServlet.java [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+@WebServlet(value = "/editProduct")
+public class EditProductServlet extends HttpServlet {
+
+    private static final ProductService productService = ProductServiceFactory.getInstance();
+    Product product;
```
Comment (Sofasmile) [inline src/main/java/controller/product/EditProductServlet.java]: can we move this field into method?

File: src/main/java/controller/user/AllUsersServlet.java [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.util.List;
+
+@WebServlet(value = "/users", loadOnStartup = 2)
+public class AllUsersServlet extends HttpServlet {
+
+    private static final UserService userService = UserServiceFactory.getInstance();
+    private List<User> allusers = userService.getAll();
```
Comment (Sofasmile) [inline src/main/java/controller/user/AllUsersServlet.java]:
```suggestion
    private List<User> allUsers = userService.getAll();
```

File: src/main/java/controller/user/EditUserServlet.java [hunk 10]
```diff
@@ -0,0 +1,79 @@
...
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+@WebServlet(value = "/editUser")
+public class EditUserServlet extends HttpServlet {
+
+    private static final UserService userService = UserServiceFactory.getInstance();
+    User user;
```
Comment (Sofasmile) [inline src/main/java/controller/user/EditUserServlet.java]: the same

File: src/main/java/dao/product/ProductDaoImpl.java [hunk 11]
```diff
@@ -0,0 +1,47 @@
...
+    }
+
+    @Override
+    public Product getById(Long id) {
+        for (Product product : PRODUCTS) {
+            if (product.getId().equals(id)) {
+                return product;
+            }
+        }
+        return null;
```
Comment (Sofasmile) [inline src/main/java/dao/product/ProductDaoImpl.java]: isn't very good idea to return null

File: src/main/java/service/user/UserServiceImpl.java [hunk 12]
```diff
@@ -0,0 +1,42 @@
...
+
+import dao.user.UserDao;
+import factory.UserDaoFactory;
+import model.User;
+
+import java.util.List;
+
+public class UserServiceImpl implements UserService {
+
+    private  static final UserDao userDao = UserDaoFactory.getInstance();
```
Comment (Sofasmile) [inline src/main/java/service/user/UserServiceImpl.java]:
```suggestion
    private static final UserDao userDao = UserDaoFactory.getInstance();
```

File: src/main/java/controller/product/EditProductServlet.java (hunk 8 above)
Comment (IgorPrihodko) [reply src/main/java/controller/product/EditProductServlet.java]: done

File: src/main/java/controller/user/AllUsersServlet.java (hunk 9 above)
Comment (IgorPrihodko) [reply src/main/java/controller/user/AllUsersServlet.java]: fixed

File: src/main/java/controller/user/EditUserServlet.java (hunk 10 above)
Comment (IgorPrihodko) [reply src/main/java/controller/user/EditUserServlet.java]: done

File: src/main/java/dao/product/ProductDaoImpl.java (hunk 11 above)
Comment (IgorPrihodko) [reply src/main/java/dao/product/ProductDaoImpl.java]: rebuilt

File: src/main/java/service/user/UserServiceImpl.java (hunk 12 above)
Comment (IgorPrihodko) [reply src/main/java/service/user/UserServiceImpl.java]: fixed

File: pom.xml [hunk 13]
```diff
@@ -3,10 +3,33 @@
...
          xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
     <modelVersion>4.0.0</modelVersion>
+    <packaging>war</packaging>
 
     <groupId>com.prihodko</groupId>
     <artifactId>shop-online</artifactId>
     <version>1.0-SNAPSHOT</version>
+    <build>
+        <plugins>
+            <!--Maven compiler plugun-->
```
Comment (Sofasmile) [inline pom.xml:13]: do we need this comments in this file?

File: src/main/java/controller/product/DeleteProductServlet.java [hunk 14]
```diff
@@ -0,0 +1,40 @@
...
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+@WebServlet(value = "/deleteProduct")
+public class DeleteProductServlet extends HttpServlet {
+
+    private static final ProductService productService = ProductServiceFactory.getInstance();
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
```
Comment (Sofasmile) [inline src/main/java/controller/product/DeleteProductServlet.java:20]: maybe better use one method to remove product?

File: src/main/java/controller/user/AllUsersServlet.java [hunk 15]
```diff
@@ -0,0 +1,27 @@
...
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.util.List;
+
+@WebServlet(value = "/users", loadOnStartup = 2)
+public class AllUsersServlet extends HttpServlet {
+
+    private static final UserService userService = UserServiceFactory.getInstance();
+    private List<User> allUsers = userService.getAll();
```
Comment (Sofasmile) [inline src/main/java/controller/user/AllUsersServlet.java:19]: better move this field into method

File: src/main/java/controller/user/DeleteUserServlet.java [hunk 16]
```diff
@@ -0,0 +1,39 @@
...
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+@WebServlet(value = "/deleteUser")
+public class DeleteUserServlet extends HttpServlet {
+
+    private static final UserService userService = UserServiceFactory.getInstance();
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
```
Comment (Sofasmile) [inline src/main/java/controller/user/DeleteUserServlet.java:20]: the same about removing

File: src/main/java/controller/user/SignInServlet.java [hunk 17]
```diff
@@ -0,0 +1,54 @@
...
+                req.setAttribute("email", email);
+            }
+            if (password.isEmpty()) {
+                req.setAttribute("error", "Password can not be empty! Try another.");
+            } else {
+                req.setAttribute("password", password);
+            }
+        }
+
+        for (User user : userService.getAll()) {
```
Comment (Sofasmile) [inline src/main/java/controller/user/SignInServlet.java:44]: better create separate method for this at service layer

File: pom.xml (hunk 13 above)
Comment (IgorPrihodko) [reply pom.xml:13]: fixed

File: src/main/java/controller/user/AllUsersServlet.java (hunk 15 above)
Comment (IgorPrihodko) [reply src/main/java/controller/user/AllUsersServlet.java:19]: fixed

File: src/main/java/controller/product/DeleteProductServlet.java (hunk 14 above)
Comment (IgorPrihodko) [reply src/main/java/controller/product/DeleteProductServlet.java:20]: i cant. i need both methods, doGet for reference in products.jsp, doPost for changing in DB.

File: src/main/java/controller/user/DeleteUserServlet.java (hunk 16 above)
Comment (IgorPrihodko) [reply src/main/java/controller/user/DeleteUserServlet.java:20]: the same as for products

File: src/main/java/controller/user/SignInServlet.java (hunk 17 above)
Comment (IgorPrihodko) [reply src/main/java/controller/user/SignInServlet.java:44]: rebuilt

**Pull request merged**